File: frontend/src/types/serviceOrder.ts
export type ServiceOrderStatus = 'Pending' | 'InProgress' | 'OnHold' | 'Completed' | 'Cancelled'
export type ServiceOrderPriority = 'Low' | 'Medium' | 'High' | 'Critical'

export interface ServiceOrder {
  id: string
  orderNumber: string
  title: string
  description: string
  status: ServiceOrderStatus
  priority: ServiceOrderPriority
  createdAt: string
  updatedAt: string
  clientName: string
  clientEmail: string
  clientPhone: string
  assignedTo: string
  estimatedCompletionDate?: string
  completedAt?: string
  notes: string
  cost: number
}

export interface ServiceOrderFormData {
  title: string
  description: string
  status: ServiceOrderStatus
  priority: ServiceOrderPriority
  clientName: string
  clientEmail: string
  clientPhone: string
  assignedTo: string
  estimatedCompletionDate?: string
  notes: string
  cost: number
}

export interface StatusCount {
  Pending: number
  InProgress: number
  OnHold: number
  Completed: number
  Cancelled: number
}

export const STATUS_LABELS: Record<ServiceOrderStatus, string> = {
  Pending: 'Pendiente',
  InProgress: 'En Progreso',
  OnHold: 'En Espera',
  Completed: 'Completado',
  Cancelled: 'Cancelado',
}

export const PRIORITY_LABELS: Record<ServiceOrderPriority, string> = {
  Low: 'Baja',
  Medium: 'Media',
  High: 'Alta',
  Critical: 'Crítica',
}

export const STATUS_OPTIONS: { value: ServiceOrderStatus; label: string }[] = [
  { value: 'Pending', label: 'Pendiente' },
  { value: 'InProgress', label: 'En Progreso' },
  { value: 'OnHold', label: 'En Espera' },
  { value: 'Completed', label: 'Completado' },
  { value: 'Cancelled', label: 'Cancelado' },
]

export const PRIORITY_OPTIONS: { value: ServiceOrderPriority; label: string }[] = [
  { value: 'Low', label: 'Baja' },
  { value: 'Medium', label: 'Media' },
  { value: 'High', label: 'Alta' },
  { value: 'Critical', label: 'Crítica' },
]

File: backend/src/SOUpgrade.Domain/Entities/ServiceOrder.cs
using SOUpgrade.Domain.Enums;

namespace SOUpgrade.Domain.Entities;

public class ServiceOrder
{
    public Guid Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ServiceOrderStatus Status { get; set; } = ServiceOrderStatus.Pending;
    public Priority Priority { get; set; } = Priority.Medium;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string ClientName { get; set; } = string.Empty;
    public string ClientEmail { get; set; } = string.Empty;
    public string ClientPhone { get; set; } = string.Empty;
    public string AssignedTo { get; set; } = string.Empty;
    public DateTime? EstimatedCompletionDate { get; set; }
    public DateTime? CompletedAt { get; set; }
    public string Notes { get; set; } = string.Empty;
    public decimal Cost { get; set; }
}
